pkg/statusreconciler/controller.go

```diff
@@ -242,8 +242,9 @@ func (c *Controller) triggerNewPresubmits(addedPresubmits map[string][]config.Pr
 				continue
 			}
 			// we want to appropriately trigger and skip from the set of identified presubmits that were
-			// added. we know all of the presubmits we are filtering need to be forced to run, so we can
-			// enforce that with a custom filter
+			// added. if a run_if_changed job has its regexp changed we should check PRs against the new regexp and
+			// trigger if needed instead of forcing the job to run on all PRs. so the default behavior of the filter need to be false.
+			// This pr https://github.com/kubernetes/test-infra/pull/12696#discussion_r284934476 led to this issue.
 			filter := pjutil.NewArbitraryFilter(func(p config.Presubmit) (shouldRun bool, forcedToRun bool, defaultBehavior bool) {
 				return true, false, false
```

Comment:
It's solved in upstream contribution: https://github.com/kubernetes-sigs/prow/pull/367

But the pull request is not updated the comments, we can contribute the updating.